M66/M66ATParser/M66ATParser.cpp

```diff
@@ -258,7 +258,9 @@ bool M66ATParser::close(int id)
     //May take a second try if device is busy
     for (unsigned i = 0; i < 2; i++) {
         if (tx("AT+QICLOSE=%d", id)
-            && scan("%d, CLOSE OK", &id_resp)) {
+            && scan("%d, CLOSE OK", &id_resp)
+            && (id == id_resp)) {
+
             return true;
```

Comment:
here if the close was successful, it tries again to close the connection, so I thought id would be better not to return (id == id_resp) so that it can try again to close the connection